recipes/OpenFst/conanfile.py

```diff
@@ -0,0 +1,146 @@
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @contextlib.contextmanager
+    def _build_context(self):
+        env = {}
+        cc, cxx = self._detect_compilers()
+        if not tools.get_env("CC"):
+            env["CC"] = cc
+        if not tools.get_env("CXX"):
+            env["CXX"] = cxx
+        with tools.environment_append(env):
+            yield
+
+    @functools.lru_cache(1)
+    def _configure_autotools(self):
+        autotools = AutoToolsBuildEnvironment(self)
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--enable-bin={}".format(yes_no(self.options.enable_bin)),
+            "--enable-compact-fsts={}".format(yes_no(self.options.enable_compact_fsts)),
+            "--enable-compress={}".format(yes_no(self.options.enable_compress)),
+            "--enable-const-fsts={}".format(yes_no(self.options.enable_const_fsts)),
+            "--enable-far={}".format(yes_no(self.options.enable_far)),
+            "--enable-fsts={}".format(yes_no(self.options.enable_fsts)),
+            "--enable-grm={}".format(yes_no(self.options.enable_grm)),
+            "--enable-linear-fsts={}".format(yes_no(self.options.enable_linear_fsts)),
+            "--enable-lookahead-fsts={}".format(yes_no(self.options.enable_lookahead_fsts)),
+            "--enable-mpdt={}".format(yes_no(self.options.enable_mpdt)),
+            "--enable-ngram-fsts={}".format(yes_no(self.options.enable_ngram_fsts)),
+            "--enable-pdt={}".format(yes_no(self.options.enable_pdt)),
+            "--enable-python={}".format(yes_no(self.options.enable_python)),
+            "--enable-special={}".format(yes_no(self.options.enable_special)),
+            "LIBS=-lpthread",
+        ]
+        if tools.get_env("CC"):
+            args.append("CC={}".format(tools.get_env("CC")))
+        if tools.get_env("CXX"):
+            args.append("CXX={}".format(tools.get_env("CXX")))
+        autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return autotools
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.libs = ["fst"]
```
Avoid collect_libs. People will need to open package/lib folder to check the libraries. Also, that method is fragile, as it uses alphabetic order to list libraries

Reply:
What if the build may have multiple libs if the options are changed? Do I need to manually add more corresponding libs when different options are enabled?